src/coreclr/nativeaot/Runtime/ThunksMapping.cpp

```diff
@@ -56,7 +57,7 @@ void EncodeThumb2Mov32(uint16_t * pCode, uint32_t value, uint8_t rDestination)
 
 COOP_PINVOKE_HELPER(int, RhpGetNumThunkBlocksPerMapping, ())
 {
-    static_assert((THUNKS_MAP_SIZE % OS_PAGE_SIZE) == 0, "Thunks map size should be in multiples of pages");
+    ASSERT_MSG((THUNKS_MAP_SIZE % OS_PAGE_SIZE) == 0, "Thunks map size should be in multiples of pages");
```

Comment:
IIRC, 64kB is a common page size on Linux Arm64. It is going to assert there.

Ask @janvorli if you need help getting Linux arm64 machine configured like that for testing.

Reply:
with `#define THUNKS_MAP_SIZE (max(0x8000, OS_PAGE_SIZE))` would it still assert?

It would still be interesting to run this on an actual system with 64K pages.

Reply:
You are right. I got it backwards.